### PR TITLE
Added more month expansions

### PR DESCRIPTION
**Pull request by bstin**
This pull request adds 2 new month expansions: MONTHSHORT and MONTHFULL. 

MONTHSHORT will print current month name in abbreviated form (e.g. Apr)
MONTHFULL will print current month name in full form (e.g. April)

These are very useful when creating templates for things like blog post, journals, letters, etc. where full month names are desired.

### CODE REVIEW

File: syntax/vim-template.vim [hunk 1]
```diff
@@ -12,7 +12,7 @@ if b:vim_template_subtype != ""
 	unlet b:current_syntax
 endif
 
-syn match vimtemplateVariable "%\%(DAY\|YEAR\|MONTH\|DATE\|TIME\|FILE\|FFILE\|EXT\|MAIL\|USER\|HOST\|GUARD\|CLASS\|MACROCLASS\|CAMELCLASS\|HERE\)%" containedin=ALL
+syn match vimtemplateVariable "%\%(DAY\|YEAR\|MONTH\|MONTHSHORT\|MONTHFULL|DATE\|TIME\|FILE\|FFILE\|EXT\|MAIL\|USER\|HOST\|GUARD\|CLASS\|MACROCLASS\|CAMELCLASS\|HERE\)%" containedin=ALL
```
Comment (aperezdc) [inline syntax/vim-template.vim]: There is a missing backslash after `MONTHFULL`, it should be:

```
 …MONTHSHORT\|MONTFULL\|DATE…
                      ↑
                   missing
```

File: plugin/templates.vim [hunk 2]
```diff
@@ -329,6 +329,8 @@ function <SID>TExpandVars()
 	let l:day        = strftime("%d")
 	let l:year       = strftime("%Y")
 	let l:month      = strftime("%m")
+	let l:monshort = strftime("%b")
+	let l:monfull  = strftime("%B")
```
Comment (aperezdc) [inline plugin/templates.vim]: Please align the equal signs (`=`) vertically.

File: plugin/templates.vim [hunk 3]
```diff
@@ -354,6 +356,8 @@ function <SID>TExpandVars()
 	call <SID>TExpand("USER",  l:user)
 	call <SID>TExpand("FDATE", l:fdate)
 	call <SID>TExpand("MONTH", l:month)
+	call <SID>TExpand("MONTHSHORT",   l:monshort)
+	call <SID>TExpand("MONTHFULL",   l:monfull)
```
Comment (aperezdc) [inline plugin/templates.vim]: Please remove the additional whitespace after the commas, so it looks like the lines below for e.g. `MACROCLASS` and `CAMELCLASS`.

**Comment by aperezdc** [comment]
This addition looks good to me, but I have commented with some nits. Could you please add the requested changes and re-upload the PR? Thank you in advance!

**Comment by bstin** [comment]
Hi aperezdc. 

Added whitespace / syntax error fixes 

Thanks for your work.

**Comment by aperezdc** [comment]
@bstin: The patch is all good now, I have merged it. Thanks for contributing!

**Pull request merged**